pkg/auth/providers/activedirectory/activedirectory_client.go

```diff
@@ -119,13 +119,24 @@ func (p *adProvider) RefetchGroupPrincipals(principalID string, secret string) (
 		return nil, err
 	}
 
-	escapedGUID := common.EscapeUUID(externalID)
-	logrus.Debugf("LDAP Refetch principals GUID : {%s}", externalID)
+	var filter string
+	var userSearchString string
+	var searchScope int
+	if strings.HasSuffix(strings.ToLower(externalID), strings.ToLower(config.UserSearchBase)) {
+		userSearchString = externalID
+		searchScope = ldapv3.ScopeBaseObject
+		filter = fmt.Sprintf("(&(%v=%v))", ObjectClass, config.UserObjectClass)
+	} else {
+		userSearchString = config.UserSearchBase
+		searchScope = ldapv3.ScopeWholeSubtree
+		escapedGUID := common.EscapeUUID(externalID)
+		filter = fmt.Sprintf("(&(%v=%v)(%v=%v))", ObjectClass, config.UserObjectClass, common.AttributeObjectGUID, escapedGUID)
+	}
 
-	filter := fmt.Sprintf("(&(%v=%v)(%v=%v))", ObjectClass, config.UserObjectClass, common.AttributeObjectGUID, escapedGUID)
+	logrus.Debugf("LDAP Refetch principals ID : {%s}", externalID)
```

Comment:
It gets ran a lot...but there has always been a similar debug line here.  I find it useful, but it may be more info that most are looking for when they turn debug on.